plugin/src/main/java/org/wildfly/plugins/bootablejar/maven/cli/ForkedCLIUtil.java

```diff
@@ -59,8 +59,15 @@ public static void fork(Log log, String[] artifacts, Class<?> clazz, Path home,
         for (String loc : artifacts) {
             cp.append(loc).append(File.pathSeparator);
         }
-        collectCpUrls(getJavaHome(), Thread.currentThread().getContextClassLoader(), cp);
-
+        StringBuilder contextCP = new StringBuilder();
+        collectCpUrls(getJavaHome(), Thread.currentThread().getContextClassLoader(), contextCP);
+        // This happens when running tests, use the process classpath to retrieve the CLIForkedExecutor main class
+        if (contextCP.toString().isEmpty()) {
+            log.warn("Re-using process classpath to retrieve Maven plugin classes to fork CLI process.");
+            cp.append(System.getProperty("java.class.path"));
+        } else {
+            cp.append(contextCP.toString());
```

Comment:
This could just be `cp.append(contextCP)`. It will invoke the `cp.append(CharSequence)` method. Again this is minor though.